subsys/bluetooth/audio/has.c

```diff
@@ -393,6 +422,54 @@ static void process_control_point_work(struct k_work *work)
 	}
 }
 
+static uint8_t get_prev_preset_index(struct has_preset *preset)
```

Comment:
This function makes me wonder if `has_preset_list` should be a double linked list (`sys_dlist_t`) instead of an array. Do we expect to do many "next" or "previous" operations? I guess this could also omit the `qsort` if we just insert the new preset in the right place in the list to keep it sorted that way.

Reply:
I've added this to notes here: https://github.com/zephyrproject-rtos/zephyr/issues/44058